src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -242,7 +243,25 @@ class GenotypeRecordReader(vcfSettings: VCFSettings) extends HtsjdkRecordReader[
   }
 }
 
-class GenericRecordReader extends HtsjdkRecordReader[Annotation] {
+
+object GenericRecordReader {
+  val haploidRegex = """^([0-9]+)$""".r
+  val diploidRegex = """^([0-9]+)([|\/]([0-9]+))$""".r
```

Comment:
I took out the "\" and the parser still worked, so I left it out. I couldn't figure out how to reference the Htsjdk constants (VCFConstants.UNPHASED...) in the regular expression string.